src/test/java/org/elasticsearch/common/geo/GeoJSONShapeParserTests.java

```diff
@@ -523,32 +511,42 @@ public void testParse_invalidPolygon() throws IOException {
                 .endArray()
                 .endObject().string();
 
-        parser = JsonXContent.jsonXContent.createParser(invalidPoly4);
+        parser = JsonXContent.jsonXContent.createParser(invalidPoly);
         parser.nextToken();
         ElasticsearchGeoAssertions.assertValidException(parser, ElasticsearchIllegalArgumentException.class);
 
         // test case 5: create an invalid polygon with 1 invalid LinearRing
-        String invalidPoly5 = XContentFactory.jsonBuilder().startObject().field("type", "polygon")
+        invalidPoly = XContentFactory.jsonBuilder().startObject().field("type", "polygon")
                 .startArray("coordinates")
                 .nullValue().nullValue()
                 .endArray()
                 .endObject().string();
 
-        parser = JsonXContent.jsonXContent.createParser(invalidPoly5);
+        parser = JsonXContent.jsonXContent.createParser(invalidPoly);
         parser.nextToken();
         ElasticsearchGeoAssertions.assertValidException(parser, ElasticsearchIllegalArgumentException.class);
 
         // test case 6: create an invalid polygon with 0 LinearRings
-        String invalidPoly6 = XContentFactory.jsonBuilder().startObject().field("type", "polygon")
+        invalidPoly = XContentFactory.jsonBuilder().startObject().field("type", "polygon")
                 .startArray("coordinates").endArray()
                 .endObject().string();
 
-        parser = JsonXContent.jsonXContent.createParser(invalidPoly6);
+        parser = JsonXContent.jsonXContent.createParser(invalidPoly);
+        parser.nextToken();
+        ElasticsearchGeoAssertions.assertValidException(parser, ElasticsearchParseException.class);
+
+        // test case 7: create an invalid polygon with 0 LinearRings
+        invalidPoly = XContentFactory.jsonBuilder().startObject().field("type", "polygon")
+                .startArray("coordinates")
+                .startArray().value(-74.011).value(40.753).endArray()
+                .endArray()
+                .endObject().string();
+
+        parser = JsonXContent.jsonXContent.createParser(invalidPoly);
         parser.nextToken();
         ElasticsearchGeoAssertions.assertValidException(parser, ElasticsearchParseException.class);
```

Comment:
It would be good to check something about the exception (like a portion of the message), so that we can be sure these aren't all hitting the exact same failure case?